### PR TITLE
Support CORS options for actuator.

### PR DESCRIPTION
**Pull request by anuraaga**
It's common to enable CORS for actuator endpoints for use from an admin console served from a different domain. This registers the CORS decorator respecting the spring configuration for it.

Also adds `Iterable` versions of all vararg methods in the CORS builders to follow our general practice.

### CODE REVIEW

**Comment by anuraaga** [comment]
Another way to think about it, it is a coincidence that `GrpcService` supports CORS, since it happens to use `doPost` and provide routes without any method restrictions. It seems weird that changing it to provide routes with `POST` restriction would cause CORS to stop working.

**Comment by trustin** [comment]
@anuraaga Perhaps it’s good time to continue tinkering with this PR now that you can so `sb.routeDecorator().catchAll().build(CorsDecorator.newDecorator(...))`? 😉

**Comment by anuraaga** [comment]
Indeed, it works!

File: spring/boot-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/CorsEndpointProperties.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.convert.DurationUnit;
+
+/**
+ * Fork of {@link org.springframework.boot.actuate.autoconfigure.endpoint.web.CorsEndpointProperties}. Upstream
+ * has a dependency on {@code CorsConfiguration} which we don't need.
```
Comment (trustin) [inline spring/boot-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/CorsEndpointProperties.java]: If this is a fork, could we 1) retain their copyright header and 2) mention which version we forked from?

Comment (anuraaga) [reply spring/boot-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/CorsEndpointProperties.java]: Oh yeah added copyright and permalink

File: core/src/main/java/com/linecorp/armeria/server/cors/AbstractCorsPolicyBuilder.java [hunk 2]
```diff
@@ -68,15 +68,15 @@
         origins = Collections.emptySet();
     }
 
-    AbstractCorsPolicyBuilder(String... origins) {
+    AbstractCorsPolicyBuilder(List<String> origins) {
         requireNonNull(origins, "origins");
-        checkArgument(origins.length > 0, "origins is empty.");
-        for (int i = 0; i < origins.length; i++) {
-            if (origins[i] == null) {
+        checkArgument(origins.size() > 0, "origins is empty.");
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/server/cors/AbstractCorsPolicyBuilder.java]: nit: `!origins.isEmpty()`

File: core/src/main/java/com/linecorp/armeria/server/cors/CorsServiceBuilder.java [hunk 3]
```diff
@@ -207,6 +221,19 @@ public CorsServiceBuilder maxAge(long maxAge) {
...
+    /**
+     * Sets the CORS {@code "Access-Control-Max-Age"} response header and enables the
+     * caching of the preflight response for the specified time. During this time no preflight
+     * request will be made.
+     *
+     * @param maxAge the maximum time that the preflight response may be cached. Rounded to seconds.
+     * @return {@link CorsServiceBuilder} to support method chaining.
+     */
+    public CorsServiceBuilder maxAge(Duration maxAge) {
+        firstPolicyBuilder.maxAge(TimeUnit.MILLISECONDS.toSeconds(maxAge.toMillis()));
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/server/cors/CorsServiceBuilder.java]: nit: usually we delegate this to `maxAge(long maxAge) {...}`?

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/server/cors/CorsServiceBuilder.java]: nit: `requireNonNull`

**Comment by anuraaga** [comment]
Thanks for cleaning up for me :)

**Comment by trustin** [comment]
Thank you, too!

**Pull request merged**